test/bin/subscription_manager_register.sh

```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env bash
+
+# Enable subscription in the VM
+subscription_manager_register() {
+    local -r vmname="$1"
+
+    if [ -f /tmp/subscription-manager-org ]; then
+        # CI workflow
+        cat <<EOF > /tmp/sub.sh
+#!/bin/bash
+set -xeuo pipefail
+
+if ! sudo subscription-manager status >&/dev/null; then
+    sudo subscription-manager register \
+        --org="\$(cat /tmp/subscription-manager-org)" \
+        --activationkey="\$(cat /tmp/subscription-manager-act-key)"
+fi
+EOF
+        copy_file_to_vm "${vmname}" /tmp/sub.sh /tmp/sub.sh
+        copy_file_to_vm "${vmname}" /tmp/subscription-manager-org /tmp/subscription-manager-org
+        copy_file_to_vm "${vmname}" /tmp/subscription-manager-act-key /tmp/subscription-manager-act-key
+        run_command_on_vm "${vmname}" "chmod +x /tmp/sub.sh && sudo /tmp/sub.sh"
+    else
+        # Local developer workflow
+        run_command_on_vm "${vmname}" "sudo subscription-manager register"
```

Comment:
Will this pause the execution and present a prompt? Could we automatically use `/etc/pki/entitlement` for RHEL hosts?

Reply:
It does pause, yes. I'm not sure what the files in that directory are. Can those just be copied to a new host as they are?

Reply:
I have a few different RHEL systems, and the contents of those files are all different. I think that means the content is host specific.

Reply:
You're right. Entitlement can be used to pass-through the subscription to a container, but in case of OS on VM it's probably not the right way.
I guess the subscription plugin system is flexible enough we can do something about these prompts if they get annoying

Reply:
I'm not sure it's possible for individual users to get the kinds of keys that we use in CI. Maybe? I think I saw something that said the account administrator had to create them.